### ISSUE TITLE
[test failed] gci-gce-alpha-features

### ISSUE THREAD

**Issue by krzyzacy** [kubernetes/kubernetes#61050]
https://k8s-testgrid.appspot.com/sig-release-master-blocking#gci-gce-alpha-features
testgrid diff for commit range: https://github.com/kubernetes/kubernetes/compare/f7d9fb474...f7aafaeb4

looks like #59862 

/priority failing-test
/priority critical-urgent
/kind bug
/status approved-for-milestone
/sig apps
/milestone v1.10

cc @jdumars @jberkus 

/assign @k82cn @janetkuo

**Comment by janetkuo** [comment]
It seems that Daemon pods aren't unable to be scheduled: 

```
I0312 14:49:59.684] Mar 12 14:49:59.525: INFO: fluentd-gcp-v3.0.0-22rr8                                                          Pending         [{PodScheduled False 0001-01-01 00:00:00 +0000 UTC 2018-03-12 14:49:39 +0000 UTC Unschedulable 0/4 nodes are available: 1 node(s) were unschedulable, 3 node(s) didn't match node selector.}]
```

**Comment by k82cn** [comment]
maybe this case: https://github.com/kubernetes/kubernetes/issues/60163#issuecomment-368773086  , let me check more detail :)

**Comment by k82cn** [comment]
ACK.  In progress, that trigger a limitation of this feature that: DaemonSet pods donot tolerate unscheduable nodes. Working on the solution.

**Comment by jdumars** [comment]
@k82cn do you have any updates on this?

**Comment by janetkuo** [comment]
I believe #61161 is tackling this issue.

**Comment by k82cn** [comment]
Working on https://github.com/kubernetes/kubernetes/pull/61161 for this :)

**Comment by k82cn** [comment]
ACK. In progress; waiting for review comments on #61161

**Comment by jberkus** [comment]
ACK. In Progress
ETA:  Later today, waiting on test results
Risks: Test may still fail.

**Comment by jberkus** [comment]
/reopen

Let's not close this until the test passes.

**Comment by krzyzacy** [comment]
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-alpha-features/19586

latest run is green :-) I'll close the issue

**Comment by krzyzacy** [comment]
🤕 @k82cn https://k8s-testgrid.appspot.com/sig-release-master-blocking#gci-gce-alpha-features it start to fail again (on before suite this time) after one green run, any insight there?

**Comment by k82cn** [comment]
@krzyzacy , I think it should be handled by #61311 .

**Comment by krzyzacy** [comment]
@k82cn ack, thanks!

**Comment by jberkus** [comment]
New PR for this issue: #61313

**Comment by krzyzacy** [comment]
(let's wait for a few green runs)

**Comment by jdumars** [comment]
This issue has more lives than a cat!

**Comment by k82cn** [comment]
/woof

:( , let me check it today

**Comment by k82cn** [comment]
ACK: in progress, open #61387 to handle a race condition, controlled by alpha feature flag

**Comment by liggitt** [comment]
after discussion with @k82cn, opened https://github.com/kubernetes/kubernetes/pull/61411 (picked to 1.10 in https://github.com/kubernetes/kubernetes/pull/61412) to disable the daemonset scheduling feature completely in 1.10 (even in the alpha job). will revisit in 1.11.

**Comment by liggitt** [comment]
Test went green on master

Waiting for cherry pick to merge before closing

**Comment by liggitt** [comment]
pick merged, watching https://k8s-testgrid.appspot.com/sig-release-1.10-blocking#gce-alpha-features-1.10

**Comment by liggitt** [comment]
https://k8s-testgrid.appspot.com/sig-release-1.10-blocking#gce-alpha-features-1.10 went green as well

/close
